### PR TITLE
Crypto Hash - The non-library version

### PR DESCRIPTION
**Pull request by arthurchan35**
Cleaned up
Made class/package naming changes according to Brian

### CODE REVIEW

File: modules/apps/portal-crypto-hash-provider/portal-crypto-hash-provider-message-digest/src/main/java/com/liferay/portal/crypto/hash/provider/message/digest/internal/factory/MessageDigestCryptoHashProvider.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
+/**
```
Comment (liferay-appsec) [inline modules/apps/portal-crypto-hash-provider/portal-crypto-hash-provider-message-digest/src/main/java/com/liferay/portal/crypto/hash/provider/message/digest/internal/factory/MessageDigestCryptoHashProvider.java:1]: When introducing a new class that implements an interface
        take into account:

* The class name should end with the interface name
* The package of the class matches the package of the interface



Refer to documentation:

File: modules/apps/portal-crypto-hash-provider/portal-crypto-hash-provider-message-digest/src/main/java/com/liferay/portal/crypto/hash/provider/message/digest/internal/factory/MessageDigestCryptoHashProviderFactory.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
+/**
```
Comment (liferay-appsec) [inline modules/apps/portal-crypto-hash-provider/portal-crypto-hash-provider-message-digest/src/main/java/com/liferay/portal/crypto/hash/provider/message/digest/internal/factory/MessageDigestCryptoHashProviderFactory.java:1]: When introducing a new class that implements an interface
        take into account:

* The class name should end with the interface name
* The package of the class matches the package of the interface



Refer to documentation:

File: modules/apps/portal-crypto-hash/portal-crypto-hash-api/configs/com.liferay.portal.crypto.hash.configuration.CryptoHashProcessorConfiguration-default.config [hunk 3]
```diff
@@ -0,0 +1 @@
+configuration.name="Default Crypto Hash Processor Configuration"
```
Comment (liferay-appsec) [inline modules/apps/portal-crypto-hash/portal-crypto-hash-api/configs/com.liferay.portal.crypto.hash.configuration.CryptoHashProcessorConfiguration-default.config:1]: It looks this pull is introducing a bundled configuration file
here.


If that is the case and this pull is fixing a bug please mark this pull as 
*DO NOT BACKPORT*


Please see https://issues.liferay.com/browse/PTR-1770

File: modules/apps/portal-crypto-hash/portal-crypto-hash-api/src/main/resources/content/Language.properties [hunk 4]
```diff
@@ -0,0 +1,7 @@
+crypto-hash-processor=Crypto Hash Processor
```
Comment (liferay-appsec) [inline modules/apps/portal-crypto-hash/portal-crypto-hash-api/src/main/resources/content/Language.properties:1]: Check that added properties are either titles or sentences

File: modules/apps/portal-crypto-hash/portal-crypto-hash-impl/src/main/java/com/liferay/portal/crypto/hash/internal/pepper/storage/DummyCryptoHashPepperStorage.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
+/**
```
Comment (liferay-appsec) [inline modules/apps/portal-crypto-hash/portal-crypto-hash-impl/src/main/java/com/liferay/portal/crypto/hash/internal/pepper/storage/DummyCryptoHashPepperStorage.java:1]: When introducing a new class that implements an interface
        take into account:

* The class name should end with the interface name
* The package of the class matches the package of the interface



Refer to documentation:

File: modules/apps/portal-crypto-hash/portal-crypto-hash-impl/src/main/java/com/liferay/portal/crypto/hash/internal/processor/CryptoHashGeneratorVerifierImpl.java [hunk 6]
```diff
@@ -0,0 +1,173 @@
+/**
```
Comment (liferay-appsec) [inline modules/apps/portal-crypto-hash/portal-crypto-hash-impl/src/main/java/com/liferay/portal/crypto/hash/internal/processor/CryptoHashGeneratorVerifierImpl.java:1]: When introducing a new class that implements an interface
        take into account:

* The class name should end with the interface name
* The package of the class matches the package of the interface



Refer to documentation:

**Review by liferay-appsec — COMMENTED** [review]
This is an automatic review suggestion

**Comment by csierra** [comment]
@arthurchan35, 

please have a look at my changes above. 
I have inlined `CryptoHashFlavor` since it seemed we did not need it anymore since we are no longer serializing it, so in can be part of the verification context directly. 

Please see my fixes for the tests, there were a lot of strings that were off. In the future please try to keep the tests working, it will save a lot of everybody's time, even though we have only one test currently it uncovered a lot of incorrect strings. 

Carlos.

**Comment by csierra** [comment]
@arthurchan35, 
please proceed to add more tests that might be relevant to the suite. Some of the ones that I commented out did not make sense anymore but some others did. 

Thx!
Carlos.

**Pull request closed without merging**